File: 03-frontend/angular-ecommerce/src/app/common/purchase.ts
import { Address }      from "./address";
import { Customer }     from "./customer";
import { Order }        from "./order";
import { OrderItem }    from "./order-item";

export class Purchase {       
    order:           Order;
    // make sure orderItems <== or any other key is the same as backend..
    // was missing the s and threw error in backend..
    orderItems:      OrderItem[];
    shippingAddress: Address;
    billingAddress:  Address;
    customer:        Customer;   
};

File: 02-backend/spring-boot-ecommerce/src/main/java/com/luv2code/ecommerce/dto/Purchase.java
package com.luv2code.ecommerce.dto;

import com.luv2code.ecommerce.entity.*;

import lombok.*;

import java.util.Set;

@Data
public class Purchase {

	private Customer customer;
	private Address	 shippingAddress;
	private Address	 billingAddress;
	private Order 	 order;
	// just a 'collection' - using REST API/JSON will take it and set to JSON Array
	private Set<OrderItem> orderItems; 
	
};
